apps/src/blockly/addons/plusMinusBlocks/advancedProcedures.js

```diff
@@ -777,34 +798,93 @@ GoogleBlockly.Extensions.register(
   'argument_report_get_var_models',
   function () {
     this.getVarModels = () => {
-      const varModels = [];
+      if (this.model) {
+        return [this.model];
+      }
+      if (this.isShadow()) {
+        return [
+          this.getRootBlock().argData_.find(
+            argDatum =>
+              argDatum.argId ===
+              this.outputConnection.targetConnection.getParentInput().name
+          ).model,
+        ];
+      }
+      // If we don't have a model yet, fall back on finding one based on the name.
       const foundModel = this.workspace.getVariable(this.getFieldValue('VAR'));
       if (foundModel) {
-        varModels.push(foundModel);
+        return [foundModel];
       }
-      return varModels;
+      console.error(`No model found for argument_reporter block`);
```

Comment:
remove this?

Reply:
This is unlikely to be hit, and my thinking was that it might be worth keeping for now. In the unlikely event that an internal user or piloter hits this, it could be helpful to know. Not a strong opinion however. FWIW, there are already a ton of places in Music Lab where information gets logged, even just during regular use.